lib/routes/account.js

```diff
@@ -323,35 +383,20 @@ module.exports = function (
             )
         }
 
-        function createResponse () {
-          if (request.query.keys !== 'true') {
-            return P.resolve({
-              sessionToken: sessionToken
-            })
+        function upsertDevice () {
+          if (! form.device) {
+            return P.resolve()
           }
 
-          var password = new Password(
-            authPW,
-            emailRecord.authSalt,
-            emailRecord.verifierVersion
-          )
-          return password.unwrap(emailRecord.wrapWrapKb)
-            .then(
-              function (wrapKb) {
-                return db.createKeyFetchToken({
-                  uid: emailRecord.uid,
-                  kA: emailRecord.kA,
-                  wrapKb: wrapKb,
-                  emailVerified: emailRecord.emailVerified
-                })
-              }
-            )
+          var operation = form.device.id ? 'updateDevice' : 'createDevice'
+
+          return db[operation](emailRecord.uid, sessionToken.tokenId, form.device)
             .then(
-              function (keyFetchToken) {
-                return {
-                  sessionToken: sessionToken,
-                  keyFetchToken: keyFetchToken
-                }
+              function (result) {
+                device = result
+              },
+              function (err) {
+                log.error({ op: 'account.login.device', err: err })
```

Comment:
For future readers: we discussed this in a meeting, agreed to disagree, and decided we're not going to be using it anyway in the initial version so we won't commit to either approach in the public API documentation just yet.